### PR TITLE
Fix circuit.load()

### PR DESCRIPTION
**Pull request by alankao64**
Bug in circuit.load().

In read_circuit_test.cpp, ReadCircuitTest, CircuitReferencingInactiveQubits:
`circuit.load(std::stringstream(kBad_Gate))` is not updating the circuit, so `circuit_data_to_tensor_network()` is getting called on `kBadTGate` for all tests.

### CODE REVIEW

**Comment by s-mandra** [comment]
@alankao64, I'm running `make -j8 && make -j8 run-tests` and everything is fine. Is this what I would expect? If not, could you adjust the test to trigger an error, so I can work from that? Thanks!

**Comment by alankao64** [comment]
> @alankao64, I'm running `make -j8 && make -j8 run-tests` and everything is fine. Is this what I would expect? If not, could you adjust the test to trigger an error, so I can work from that? Thanks!

Everything should pass, the reason I found this bug is because we wanted to add the gate into the error message, and then I found that circuit.load() wasn't actually updating circuit. I'll adjust the test to fail.

**Comment by alankao64** [comment]
@s-mandra the test should be failing now!

**Comment by s-mandra** [comment]
Everything works now. Thanks for catching this @alankao64!

File: tests/src/read_circuit_test.cpp [hunk 1]
```diff
@@ -103,9 +103,7 @@ constexpr char kBadTGate[] = R"(1
 constexpr char kBadCzGate[] = R"(1
 1 cz 0 1)";
 
-constexpr char kBadCxGate[] = R"(1
-1 cx 0 1)";
-
+// Broken test
```
Comment (95-martin-orion) [inline tests/src/read_circuit_test.cpp]: Is this test still broken?

File: src/read_circuit.cpp [hunk 2]
```diff
@@ -395,7 +405,9 @@ void circuit_data_to_tensor_network(
       // Check that position is an active qubit
       bool qubit_off = find_grid_coord_in_list(off, i_j_1[0], i_j_1[1]);
       if (qubit_off)
-        throw ERROR_MSG("Qubit ", q1, " must correspond to an active qubit.");
+        throw ERROR_MSG("Qubit '", q1, "' in gate '", gate.cycle, " ",
```
Comment (95-martin-orion) [inline src/read_circuit.cpp]: If possible, these should use a method in QflexGate to print the gate - maybe store the exact string for each gate when loading the circuit? The log here will skip `gate.params` if it has any.

Same for 2-qubit gates.

File: tests/src/read_circuit_test.cpp (hunk 1 above)
Comment (s-mandra) [reply tests/src/read_circuit_test.cpp]: No, everything should work. Let me remove the comment.

File: src/read_circuit.cpp (hunk 2 above)
Comment (s-mandra) [reply src/read_circuit.cpp]: Let me take a look into it.

Comment (s-mandra) [reply src/read_circuit.cpp]: Fixed.

**Pull request merged**